lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -75,3 +75,122 @@ function _wp_theme_has_theme_json_clean_cache_upon_upgrading_active_theme( $upgr
 		}
 	}
 }
+
+/**
+ * Returns the stylesheet resulting of merging core, theme, and user data.
+ *
+ * @param array $types Types of styles to load. Optional.
+ *                     It accepts 'variables', 'styles', 'presets' as values.
+ *                     If empty, it'll load all for themes with theme.json support
+ *                     and only [ 'variables', 'presets' ] for themes without theme.json support.
+ *
+ * @return string Stylesheet.
+ */
+function gutenberg_get_global_stylesheet( $types = array() ) {
+	/**
+	 * Filters whether the cached global stylesheet can be used.
+	 *
+	 * @param boolean $can_use_cached Whether the cached global stylesheet can be used.
+	 */
+	$can_use_cached = apply_filters(
+		'wp_get_global_stylesheet_can_use_cache',
+		( empty( $types ) ) &&
+		// Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
+		( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG )
```

Comment:
I believe we removed this form #45543. I'm personally still in favor of having it present, but we should be consistent about this. I would suggest to remove this from here for now and open a separate issue/PR to holistically revisit: **How do we best support the theme developer workflow and bypass cache for this kind of thing?** Maybe `WP_DEBUG` is the answer, maybe a new constant, maybe something else.

Reply:
I agree we have to use `WP_DEBUG` in this PR and in the [settings PR](https://github.com/WordPress/gutenberg/pull/45372), as styles and settings do change while developing a new theme or working with `theme.json` options.

For consistency, it'd be great to also have it https://github.com/WordPress/gutenberg/pull/45543 Though I find that use case a lot less impactful and uncommon (it covers the case a theme changes whether it has a `theme.json` or not). Happy to add it back there in a different PR unless we come up with a different alternative (asked [here](https://github.com/WordPress/gutenberg/pull/45679#discussion_r1020481866), but none was provided so far).

Reply:
@oandregal FWIW I'm still in favor of adding it back, so 👍 from me on that. Agreed that the likeliness to be relevant there is less critical since a dev would barely ever add/remove `theme.json` during development, but I think still useful for consistency since in principle it is subject to similar conditions as any change to `theme.json`.

Reply:
I'm also ok adding in back for `wp_theme_has_theme_json`. I've only removed it to unblock that PR from landing, given it was not essential.

Reply:
I created https://github.com/WordPress/gutenberg/pull/45882 to add back the `WP_DEBUG` check to `wp_theme_has_theme_json`. I'm happy to remove the `WP_DEBUG` check from here and add it to that new PR instead in case we want to further discuss whether we should use something else (but I honestly think that checking `WP_DEBUG` is the good to go, since it works out of the box as opposed to a new constant or filter which would require theme developers to learn and configure them).

Reply:
We shouldn't remove `WP_DEBUG` from this PR, unless we find an alternative way to provide the same developer experience out of the box. It is a pre-existing condition (not added by this PR).